bochs/cpu/flag_ctrl_pro.cc

```diff
@@ -129,25 +129,32 @@ void BX_CPU_C::handleInterruptMaskChange(void)
     }
 #endif
 
+#if BX_SUPPORT_UINTR
+    if (!uintr_masked()) unmask_event(BX_EVENT_PENDING_UINTR);
+#endif
+
     return;
   }
 
   // EFLAGS.IF was cleared, some events like INTR would be masked
 
 #if BX_SUPPORT_VMX
   if (BX_CPU_THIS_PTR in_vmx_guest && PIN_VMEXIT(VMX_VM_EXEC_CTRL1_EXTERNAL_INTERRUPT_VMEXIT)) {
-     // if 'External-interrupt exiting' control is set, the value of EFLAGS.IF
-     // doesn't affect interrupt blocking
-     mask_event(BX_EVENT_VMX_INTERRUPT_WINDOW_EXITING | BX_EVENT_PENDING_VMX_VIRTUAL_INTR);
-     unmask_event(BX_EVENT_PENDING_INTR | BX_EVENT_PENDING_LAPIC_INTR);
+    // if 'External-interrupt exiting' control is set, the value of EFLAGS.IF
+    // doesn't affect interrupt blocking
+    mask_event(BX_EVENT_VMX_INTERRUPT_WINDOW_EXITING | BX_EVENT_PENDING_VMX_VIRTUAL_INTR);
+    unmask_event(BX_EVENT_PENDING_INTR | BX_EVENT_PENDING_LAPIC_INTR);
+#if BX_SUPPORT_UINTR
+    if (!uintr_masked()) unmask_event(BX_EVENT_PENDING_UINTR);
+#endif
      return;
   }
 #endif
 
 #if BX_SUPPORT_SVM
   if (BX_CPU_THIS_PTR in_svm_guest && SVM_V_INTR_MASKING) {
      if (! SVM_HOST_IF)
-       mask_event(BX_EVENT_PENDING_INTR | BX_EVENT_PENDING_LAPIC_INTR);
+       mask_event(BX_EVENT_PENDING_INTR | BX_EVENT_PENDING_LAPIC_INTR | BX_EVENT_PENDING_UINTR);
```

Comment:
I thought UINTR is an Intel-only technology, how could it affect SVM-specific code?

Reply:
I bet AMD will copy it soon so enabled it already as well as they usually do.
User-level interrupt is being asked by customers from both companies for very long time